tests/rptest/tests/tiered_storage_model_test.py

```diff
@@ -568,32 +571,35 @@ def test_tiered_storage(self, cloud_storage_type_and_url_style: List[
             test_case_name = json.loads(test_case)["name"]
             test_case = get_test_case_from_name(test_case_name)
             assert test_case is not None, f"no test case found with name {test_case_name}"
-        # Configuration phase
-        self.start_inputs(test_case)
-        self.start_validators(test_case)
-        self.run_bg_validators(test_case)
 
-        # Setup stage
-        self.prepare_stage(TestRunStage.Startup, test_case)
-        self.run_stage_validators(TestRunStage.Startup, test_case)
+        try:
+            # Configuration phase
+            self.start_inputs(test_case)
+            self.start_validators(test_case)
+            self.run_bg_validators(test_case)
 
-        # Producing the data
-        self.prepare_stage(TestRunStage.Produce, test_case)
-        self.produce_until_validated(test_case)
+            # Setup stage
+            self.prepare_stage(TestRunStage.Startup, test_case)
+            self.run_stage_validators(TestRunStage.Startup, test_case)
 
-        # Intermediate step
-        self.prepare_stage(TestRunStage.Intermediate, test_case)
-        self.run_stage_validators(TestRunStage.Intermediate, test_case)
+            # Producing the data
+            self.prepare_stage(TestRunStage.Produce, test_case)
+            self.produce_until_validated(test_case)
 
-        # Consuming the data
-        self.prepare_stage(TestRunStage.Consume, test_case)
-        self.consume_until_validated(test_case)
+            # Intermediate step
+            self.prepare_stage(TestRunStage.Intermediate, test_case)
+            self.run_stage_validators(TestRunStage.Intermediate, test_case)
 
-        # Shutting down
-        self.prepare_stage(TestRunStage.Shutdown, test_case)
-        self.run_stage_validators(TestRunStage.Shutdown, test_case)
+            # Consuming the data
+            self.prepare_stage(TestRunStage.Consume, test_case)
+            self.consume_until_validated(test_case)
 
-        self.shutdown_bg_tasks()
+            # Shutting down
+            self.prepare_stage(TestRunStage.Shutdown, test_case)
+            self.run_stage_validators(TestRunStage.Shutdown, test_case)
+
+        finally:
+            self.shutdown_bg_tasks("test ending")
```

Comment:
The idea is that once you leave the function, via exception, this was skipped and then we exit the @cluster wrapper which stops Redpanda, and the BG tasks are still running. Then after, in `tearDown` they are stopped, but the log reader task may be hung because it's doing `tail -f` on a file that will never get another line.